### DISCUSSION TITLE
API Payload Documentation

### DISCUSSION THREAD

**Discussion by noaheverett** [cloud-hypervisor/cloud-hypervisor#4828] [Q&A] [answered]
I'm reading about the API at https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/api.md and trying to find where the request payloads are documented/defined.

Ex when creating a VM the payload request is shown in the API docs as /schemas/VmConfig or /schemas/VmResize for VM resize.

Is there a document showing the JSON payload/params for each API endpoint?

Apologies if I've missed, I've tried searching the source code etc for references to the API and /schemas/*

It's probably somewhere and I'm missing it, if so I apologize in advance! 🤦‍♂️

**Comment by rbradford** [comment, marked as answer]
This might be the missing piece you need: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/vmm/src/api/openapi/cloud-hypervisor.yaml#L515

**Reply by noaheverett** [reply]
Apparently I need to work on my Github code searching skills! 😆 Exactly what I need, thank you
